dm/dm/master/server.go

```diff
@@ -616,13 +616,14 @@ func (s *Server) OperateTask(ctx context.Context, req *pb.OperateTaskRequest) (*
 	case pb.TaskOp_Resume:
 		expect = pb.Stage_Running
 	case pb.TaskOp_Delete:
-		expect = pb.Stage_Stopped
+		// op_delete means delete this running subtask, we not have stage_null now, so use invalid stage instead
+		expect = pb.Stage_InvalidStage
 	default:
 		resp.Msg = terror.ErrMasterInvalidOperateOp.Generate(req.Op.String(), "task").Error()
 		return resp, nil
 	}
 	var err error
-	if expect == pb.Stage_Stopped {
+	if expect == pb.Stage_InvalidStage {
```

Comment:
or just check `op ==  pb.TaskOp_Delete` here, `Stage_InvalidStage` is still misleading

Reply:
added a new stage null , seems this is better https://github.com/pingcap/tiflow/pull/4510/commits/24001949601d41d7e8a4497422bcd143d34aaeb5